File: kimdokyun/Python/boj_1074.py
def choose(N,r,c):
    if N==0: return 0
    else:
        cutLine = 2**(N-1)
        if r < cutLine and c < cutLine:
            return choose(N-1,r,c)
        elif r < cutLine and c >= cutLine:
            return cutLine**2 + choose(N-1,r,c-cutLine)
        elif r >= cutLine and c < cutLine:
            return 2*(cutLine**2) + choose(N-1,r-cutLine,c)
        else:
            return 3*(cutLine**2) + choose(N-1,r-cutLine,c-cutLine)
N,r,c = map(int,input().split())
print(choose(N,r,c))

File: kimdokyun/C++/boj_1074.cpp
#include<stdio.h>
#include<math.h>

int choose(int N, int r, int c) {
	if(N==0) return 0;
	else {
		int cutLine = pow(2,N-1);
		if(r<cutLine && c<cutLine) {
			return choose(N-1,r,c);
		}
		else if(r<cutLine && c>=cutLine) {
			return pow(cutLine,2) + choose(N-1,r,c-cutLine);
		}
		else if(r>=cutLine && c<cutLine) {
			return 2*pow(cutLine,2) + choose(N-1,r-cutLine,c);
		}
		else {
			return 3*pow(cutLine,2) + choose(N-1,r-cutLine,c-cutLine);
		}
	}
}

int main() {
	int N,r,c;
	scanf("%d %d %d",&N,&r,&c);
	printf("%d",choose(N,r,c));
}
